neigh_test.go

```diff
@@ -257,26 +257,30 @@ func TestNeighAddDelProxy(t *testing.T) {
 	}
 }
 
-func expectNeighUpdate(ch <-chan NeighUpdate, t uint16, state int, ip net.IP) error {
+func expectNeighUpdate(ch <-chan NeighUpdate, t uint16, state int, ip net.IP) (err error) {
 	for {
 		timeout := time.After(time.Second)
 		select {
 		case update := <-ch:
 			if update.Type != t {
-				return fmt.Errorf("want %d got %d", t, update.Type)
+				err = fmt.Errorf("want update type %d got %d", t, update.Type)
 			}
 			if update.Neigh.State != state {
-				return fmt.Errorf("want %d got %d", state, update.State)
+				err = fmt.Errorf("want neight state %d got %d", state, update.State)
 			}
 			if update.Neigh.IP == nil {
-				return fmt.Errorf("Neigh.IP is nil")
+				err = fmt.Errorf("Neigh.IP is nil")
 			}
 			if !update.Neigh.IP.Equal(ip) {
-				return fmt.Errorf("Neigh.IP: want %s got %s", ip, update.Neigh.IP)
+				err = fmt.Errorf("Neigh.IP: want %s got %s", ip, update.Neigh.IP)
 			}
-			return nil
+			err = nil
+			return
 		case <-timeout:
-			return fmt.Errorf("timeout")
+			if err == nil {
+				err = fmt.Errorf("timeout")
+			}
+			return
```

Comment:
What does using a naked return give us here ?
Besides the error msg improvement, I'd prefer the direct `return fmt.Errorf()` on error.